genesyscloud/conversations_messaging_settings_default/resource_conversations_messaging_settings_default.go

```diff
@@ -0,0 +1,86 @@
+
+	log.Printf("Reading conversations messaging settings default %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		messagingSettingDefault, resp, getErr := proxy.getConversationsMessagingSettingsDefault(ctx)
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read conversations messaging settings default %s: %s", d.Id(), getErr), resp))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read conversations messaging settings default %s: %s", d.Id(), getErr), resp))
+		}
+
+		resourcedata.SetNillableValue(d, "setting_id", messagingSettingDefault.Id)
+
+		log.Printf("Read conversations messaging settings default %s %s", d.Id(), *messagingSettingDefault.Id)
+		return cc.CheckState(d)
+	})
+}
+
+// updateConversationsMessagingSettingsDefault is used by the conversations_messaging_settings_default resource to update an conversations messaging settings default in Genesys Cloud
+func updateConversationsMessagingSettingsDefault(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getConversationsMessagingSettingsDefaultProxy(sdkConfig)
+	settingId := d.Get("setting_id").(string)
+
+	updateRequest := platformclientv2.Messagingsettingdefaultrequest{
+		SettingId: &settingId,
+	}
+
+	log.Printf("Updating conversations messaging settings default %s", settingId)
+	messagingSettingDefault, resp, err := proxy.updateConversationsMessagingSettingsDefault(ctx, &updateRequest)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update conversations messaging settings default %s: %s", d.Id(), err), resp)
+	}
+
+	log.Printf("Updated conversations messaging settings default %s", *messagingSettingDefault.Id)
+	return readConversationsMessagingSettingsDefault(ctx, d, meta)
+}
+
+func deleteConversationsMessagingSettingsDefault(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getConversationsMessagingSettingsDefaultProxy(sdkConfig)
+
+	log.Printf("Deleting conversations messaging settings default")
+	resp, err := proxy.deleteConversationsMessagingSettingsDefault(ctx)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to delete conversations messaging settings default %s: %s", d.Id(), err), resp)
+	}
+
+	return util.WithRetries(ctx, 180*time.Second, func() *retry.RetryError {
+		defaultSetting, resp, err := proxy.getConversationsMessagingSettingsDefault(ctx)
+		if err != nil || defaultSetting == nil {
+			log.Printf("Deleted conversations messaging settings default %s", d.Id())
+			return nil
+		}
```

Comment:
I noticed the API returns a 200 if no default settings exist, which is kind of weird behaviour. In case they ever clean up their API, I would extend this condition to look something like 

```golang
if err != nil {
  if util.IsStatus404(resp) {
    // it was deleted
    return nil  
  }
  return non retryable error
}
if defaultSetting == nil {
  // it was deleted
  return nil
}